pulsar-rpc-contrib/src/main/java/org/apache/pulsar/rpc/contrib/common/MessageDispatcherFactory.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.pulsar.rpc.contrib.common;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.regex.Pattern;
+import lombok.RequiredArgsConstructor;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.ConsumerBuilder;
+import org.apache.pulsar.client.api.MessageListener;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.ProducerAccessMode;
+import org.apache.pulsar.client.api.ProducerBuilder;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SubscriptionInitialPosition;
+import org.apache.pulsar.client.api.SubscriptionType;
+
+/**
+ * Facilitates the creation of producers and consumers for Pulsar based on specified schemas and configurations.
+ * This factory simplifies the setup of Pulsar clients for sending and receiving messages in RPC context.
+ *
+ * @param <T> the type parameter for the request messages.
+ * @param <V> the type parameter for the reply messages.
+ */
+@RequiredArgsConstructor
+public class MessageDispatcherFactory<T, V> {
+    private final PulsarClient client;
+    private final Schema<T> requestSchema;
+    private final Schema<V> replySchema;
+    private final String subscription;
+
+    /**
+     * Creates a Pulsar producer for sending requests. (Pulsar RPC Client side)
+     *
+     * @param requestProducer this is the producer that the user can pass in that contains additional configuration.
+     * @return the created request message producer.
+     * @throws IOException if there is an error creating the producer.
+     */
+    public Producer<T> requestProducer(ProducerBuilder<T> requestProducer) throws IOException {
+        return requestProducer
+                // allow only one client
+                .accessMode(ProducerAccessMode.Exclusive)
```

Comment:
We can expose a way to modify the Producer and consumer configuration by `.loadConf()`.

Reply:
Each message and the request producer has been modified using the `loadConf()`.